### PR TITLE
swaybar: use height from config

### PR DESCRIPTION
**Pull request by crondog**
I had to move the window_setup code around. I don't think it breaks anything...

### CODE REVIEW

File: swaybar/main.c [hunk 1]
```diff
@@ -466,6 +477,12 @@ int main(int argc, char **argv) {
 
 	int desired_output = atoi(argv[optind]);
 	struct output_state *output = registry->outputs->items[desired_output];
+	
+	window = window_setup(registry, output->width, 30, false);
```
Comment (mikkeloscar) [inline swaybar/main.c]: I assume the height parameter here is relevant as well.

**Comment by mikkeloscar** [comment]
We also really need to split up swaybar into multiple files at some point as it's becoming a bit of a mess.

File: swaybar/main.c (hunk 1 above)
Comment (crondog) [inline swaybar/main.c]: I wasn't really sure how to do this, as we need the window created before we can find the height of the text when no height is specified. The 30 being passed here is overwriten later before the window gets rendered anyway so it doesn't matter. It's a bit of a catch 22. I suppose I can change it to 0 to make it cleaner

**Comment by crondog** [comment]
Should there be a swaybar/config.c file which loads in all of the configs or something else?

File: swaybar/main.c (hunk 1 above)
Comment (mikkeloscar) [inline swaybar/main.c]: I see, I think it it more clear with 0.

**Comment by mikkeloscar** [comment]
Yes, I think we should have something like a `struct bar_config` which is initialized from the json data in a separate function/file, so we don't mix parsing of the json with the initialization of the window for instance. We could also refactor the ipc stuff to a separate file.

@SirCmpwn where should we put *local* header files only relevant to swaybar? Should they just go in `inlcude/` or maybe in something like `swaybar/include`?

File: swaybar/main.c (hunk 1 above)
Comment (SirCmpwn) [inline swaybar/main.c]: Nah, leave it be. 0 is going to cause issues.

**Comment by SirCmpwn** [comment]
@mikkeloscar I'll be refactoring the crap out of swaybar et al soon.

**Comment by SirCmpwn** [comment]
Thanks @crondog!

**Pull request merged**